[BE-3152]: Added FOD docs (#429)
appcircleio/appcircle-docusaurus: docs/workflows/common-workflow-steps/fod-mobile-assesment.md

---
title: Fortify on Demand Mobile Assessment
description: Optimize mobile app security with Fortify on Demand Mobile Assessment. Comprehensive testing for robust, secure mobile applications.
tags: [mobile, mobile ci/cd, security, integration, ios, android]
sidebar_position: 4
---

import Screenshot from '@site/src/components/Screenshot';

# Fortify on Demand Mobile Assessment

[Fortify on Demand Mobile Assessment](https://www.opentext.com/products/fortify-on-demand) provides a robust solution for securing mobile applications by identifying vulnerabilities before they can be exploited. This comprehensive cloud-based service streamlines the security testing process, ensuring your mobile apps meet the highest standards of security.

> Fortify on Demand Mobile Assessment can be seamlessly integrated into your workflow with Appcircle, enabling easy setup and utilization within your existing development processes.

Reviewer: ```suggestion
You can seamlessly integrate Fortify on Demand Mobile Assessment into your workflow with Appcircle, facilitating easy setup and utilization within your existing development processes.
```

> ### Prerequisites

Reviewer 2: Flutter iOS and Android should be added too. 
The Prerequisites table should be under each platform heading.
You can examine it as an example:
- https://docs.appcircle.io/workflows/common-workflow-steps/firebase-app-distribution#prerequisites

> Before running the **Fortify on Demand Mobile Assessment** step, certain prerequisites must be completed. These prerequisites, detailed in the table below:

Reviewer: ```suggestion
Before running the **Fortify on Demand Mobile Assessment** step, you must complete certain prerequisites, as detailed in the table below:
```

| Prerequisite Workflow Step                      | Description                                     |
|-------------------------------------------------|-------------------------------------------------|
| [**Xcodebuild for Devices**](/workflows/ios-specific-workflow-steps/xcodebuild-for-devices) | This step will build your application in ARM architecture and generate an `IPA` and `dSYM` file. |
| [**Android Build**](/workflows/android-specific-workflow-steps/android-build) | The app required for this step is generated by the **Android Build** (or alternative build steps). |

> | [**Android Sign**](/workflows/android-specific-workflow-steps/android-sign) | If a signed app is created, this step needs to be run beforehand to process this output.|

Reviewer: ```suggestion
| [**Android Sign**](/workflows/android-specific-workflow-steps/android-sign) | If you create a signed app, you must run this step beforehand to process the output. |
```

#### For Android

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE3152-fodAndroidOrder.png' />

### Input Variables

Below is a list of input variables that can be used with this component with a description of each.

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE3152-fodInput.png' />

:::warning

Avoid hard-coding sensitive information like tokens and API keys directly into the step parameters.

We recommend using [**Environment Variables**](/environment-variables/managing-variables) groups for such sensitive variables.

:::

| Variable Name                 | Description                                                                                               | Status   |
|-------------------------------|-----------------------------------------------------------------------------------------------------------|----------|
| `$AC_FOD_CLIENT_ID`           | Authenticating with Client Credentials. The client credentials are the API keys generated in the portal   | Required |
| `$AC_FOD_CLIENT_SECRET`       | Authenticating with Client Credentials. The client credentials are the API keys generated in the portal.  | Required |

> | `$AC_FOD_CLIENT_DATACENTER`   | Select the datacenter assigned to you. This is provided once your are onboarded with Fortify on Demand.   | Required |

Reviewer: ```suggestion
| `$AC_FOD_CLIENT_DATACENTER`   | Select the datacenter assigned to you upon your onboarding with Fortify on Demand.   | Required |
```

> | `$AC_FOD_ENTITLEMENT_ID`      | The ID assigned to ypur current entitlement. This value can be reterived in Foritfy on Demand portal.     | Required |

Reviewer: ```suggestion
| `$AC_FOD_ENTITLEMENT_ID`      | The ID assigned to your current entitlement. You can retrieve this value in the Fortify on Demand portal.     | Required |
```
